From 4c27b42e63349b5d6ef2c798a9d2026b1b32979d Mon Sep 17 00:00:00 2001
From: Roberto Granados <x0095451@ti.com>
Date: Thu, 20 Jan 2011 19:50:07 -0600
Subject: OMAP4: HDMI: Fix order dependency for audio

Audio driver was assuming video driver would be initialized before
it initilizes itself, this was causing problems since the mutex on
hdmi_lib.c hadn't been initialized.
CSR/DR: OMAPS00230459/OMAPS00231115

Change-Id: Ide63ad79f255e1a7716e47bd815977ece5afec9c
Signed-off-by: Roberto Granados Dorado <x0095451@ti.com>
---
 arch/arm/plat-omap/hdmi_lib.c |   22 ++++++++++++++++------
 sound/soc/omap/omap-hdmi.c    |    1 +
 2 files changed, 17 insertions(+), 6 deletions(-)

NOTE: arch/arm/plat-omap/hdmi_lib.c changes are already taken care

diff --git a/sound/soc/omap/omap-hdmi.c b/sound/soc/omap/omap-hdmi.c
index 454b2ee..2ba7fd3 100644
--- a/sound/soc/omap/omap-hdmi.c
+++ b/sound/soc/omap/omap-hdmi.c
@@ -242,6 +242,7 @@ static __devinit int omap_hdmi_probe(struct platform_device *pdev)
 	notifier->hpd_notifier = hdmi_hpd_notifier;
 	notifier->pwrchange_notifier = hdmi_pwrchange_notifier;
 	notifier->private_data = &hdmi_data;
+	hdmi_lib_init();
 	hdmi_add_notifier(notifier);
 
 	return snd_soc_register_dai(&pdev->dev, &omap_hdmi_dai);
-- 
1.6.0.4

